static/docs/commands-reference/commit.md

```diff
@@ -2,40 +2,212 @@
+  With this option, `dvc commit` determines the files to commit by searching the
+  named directory, and its subdirectories, for DVC files for which to commit
+  data. Along with providing a `target`, or `target` along with `--with-deps`,
+  it is yet another way to limit the scope of DVC files to upload.
+
+* `-f`, `--force` - commit data even if checksums for dependencies or outputs
+  did not change.
+
 * `-h`, `--help` - prints the usage/help message, and exit.
 
-* `-q`, `--quiet` - does not write anything to standard output. Exit with 0 if
-  no problems arise, otherwise 1.
+* `-q`, `--quiet` - do not write anything to standard output. Exit with 0 if all
+  stages are up to date or if all stages are successfully rerun, otherwise exit
+  with 1.
+
+* `-v`, `--verbose` - displays detailed tracing information from executing the
+  `dvc add` command.
 
-* `-v`, `--verbose` - displays detailed tracing information.
 
 ## Examples
 
+Let's start with the example workspace used in the
+[Getting Started](/doc/get-started) tutorial.  Setup requires these steps:
+
+```dvc
+    $ git clone https://github.com/iterative/example-get-started
+    $ cd example-get-started
+    $ dvc pull
+```
+
+The workspace in the Git repository is preconfigured with a DVC remote containing
+precomputed data we can use in the following examples.
+
+## Example: Editing a data file
+
+Sometimes we want to edit a data or configuration file and rerun the pipeline to
+see what difference it makes.  To replace a data file we'd use the `dvc remove`
+command, but to edit it we use `dvc unprotect`.
+
+```dvc
+    $ ls -l data/
+    total 74072
+    -rw-r--r--  2 david  admin  37916850 Apr 11 17:03 data.xml
+
+    $ dvc unprotect data/data.xml
+    [##############################] 100% Unprotecting 'data/data.xml'
+
+    $ ls -l data/
+    total 74072
+    -rw-r--r--  1 david  admin  37916850 Apr 11 17:06 data.xml
+```
+
+The `1` in link count here indicates DVC has unprotected the file.  For other
+link types this will appear in different ways.  At this point we can safely edit
+the data file.
+
+After editing the file we can use `dvc commit` to update the changes.
+
+```dvc
+    $ dvc commit
+    ...
+    outputs ['data/data.xml'] of 'data/data.xml.dvc' changed. Are you sure you commit it? [y/n] y
+    ...
+    dependencies ['data/data.xml'] of 'prepare.dvc' changed. Are you sure you commit it? [y/n] y
+```
+
+Because this file is referenced from two DVC files, we are queried twice.
+
+## Example: Running the pipeline without committing data changes
+
+As was suggested in the previous section, sometimes we want to change the data
+to try different options.  To avoid filling the cache with temporary results as
+you try different algorithms we can unprotect all affected files, and use the
+`dvc repro --no-commit` option to prevent the cache from being updated.
+
+We start with unprotecting the files involved in the affected stages:
```

Comment:
`dvc repro` is removing outputs (or unprotecting them if they are persistent), no need to run unprotect in this case

Reply:
Oh... `dvc repro --no-commit` will do the equivalent of unprotect as a byproduct?

Reply:
Yes, the only file that must be unprotected in that example is data/data.xml and the other files end up uncommitted at the end.

Reply:
I'm not sure I understand about "other files". `dvc repro` and `dvc run` *always* remove (or at least unprotect if options to keep output are specified) outputs. No need to unprotect anything at all. Only input data set at the very top if edit it manually (which I think it not a good example by itself to be honest - it's better to iterate with code - the way we do on get started, the way I suggested here already).